server/src/organizations/organizations.service.ts

```diff
@@ -1,19 +1,35 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
 import { CreateOrganizationDto } from './dto/create-organization.dto';
 import { UpdateOrganizationDto } from './dto/update-organization.dto';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository, DeleteResult } from 'typeorm';
 import { Organization } from './entities/organization.entity';
+import fetch from 'node-fetch';
 
 @Injectable()
 export class OrganizationsService {
   constructor(
-    @InjectRepository(Organization)
-    private organizationsRepository: Repository<Organization>,
+    @InjectRepository(Organization) private organizationsRepository: Repository<Organization>,
   ) {}
 
   async create(createOrganizationDto: CreateOrganizationDto): Promise<Organization> {
-    return await this.organizationsRepository.save(createOrganizationDto);
+    return this.organizationsRepository.save(createOrganizationDto);
+  }
+
+  async checkEIN(ein: number): Promise<object> {
+    try {
+      const res = await fetch(
+        `https://projects.propublica.org/nonprofits/api/v2/organizations/${ein}.json`,
+      );
+      const org = await res.json();
+      if (org) {
+        return org.organization;
+      } else {
+        return null
+      }
+    } catch (err) {
+      console.log(err.message)
+    }
```

Comment:
i think we should also have a return of some sort in here
not sure what makes the most sense
maybe something like
```
if err === not found error
  return err; // (basically throw 404 (expected response if propublica org doesn't exist)
else
  // idk, maybe we want to throw all the time?
  // remind me what the propublica api responds with again
```

Reply:
actually, if it responds with 404 when we also want to respond with that, we may not need the try/catch and might just expose this error as our response too

Reply:
if the best option is unclear we can always fix when we do the FE ticket

Reply:
we actually do want a response, otherwise our FE would just be waiting in some loading state or something
so it should return a 404 or something and then on FE we can handle as needed

<img width="992" alt="Screen Shot 2022-01-08 at 1 00 15 PM" src="https://user-images.githubusercontent.com/16400333/148659856-f4bd945d-9060-4fff-88f0-f77b772eb83d.png">

Reply:
also not `invalid json` error in server console when bad ein is passed in

Reply:
Hey, sorry I know it's been a while:

Would this work? 

`    } catch (err) {
      return err;
    }`

The value is then: 
`{
    "message": "invalid json response body at https://projects.propublica.org/nonprofits/api/v2/organizations/142007221.json reason: Unexpected token < in JSON at position 0",
    "type": "invalid-json"
}`

Not sure what this would throw us on the front end, but at least it's something for now?

Reply:
Or is the goal to find out how to turn it into a 404?

Reply:
cause I could throw a NotFoundException and that would return a 404. Maybe that's the move.

Reply:
Okay @jd2rogers2 I went for the NotFoundException route. If that work for you I think it's ready to merge.  I made the other changes if you want to double check!